### PR TITLE
Deployment Settings for interval in S3.end_date_interval

### PR DESCRIPTION
**Pull request by gauravmittal1995**
Please let me know which ones need explicit default.
Any other suggestions?

### CODE REVIEW

**Comment by gauravmittal1995** [comment]
@flavour Please review

**Comment by flavour** [comment]
Travis CI is failing for some reason.
In general I'm very wary about this - seems like we're adding a lot of complexity for little value?
We already have too many deployment_settings really...I suspect these will be very unlikely to be set & so are just more settings that people have to wade through & try to understand when they're trying to configure their system.
Just because we can, doesn't necessarily mean that we should?

**Comment by gauravmittal1995** [comment]
@flavour Removed the deployment Settings. Should all these have implicit default? or some of them should have explicit defaults??

**Comment by gauravmittal1995** [comment]
@flavour Also is there any important instance in one of these where a deployment setting may come useful?

File: modules/s3db/gis.py [hunk 1]
```diff
@@ -242,6 +242,8 @@ def model(self):
                     ),
             s3_date("end_date",
                     label = T("End Date"),
+                    start_field = "gis_location_start_date",
+                    default_interval = 12,
```
Comment (flavour) [inline modules/s3db/gis.py]: I don't think this has a default_interval at all...Locations change based on local politics

File: modules/s3db/survey.py [hunk 2]
```diff
@@ -1633,6 +1633,8 @@ def model(self):
                                   ),
                           s3_date("end_date",
                                   label = T("End Date"),
+                                  start_field = "survey_series_start_date",
+                                  default_interval = 12,
```
Comment (flavour) [inline modules/s3db/survey.py]: A survey_series is normally a very brief time...default_interval might be 1 month

**Comment by gauravmittal1995** [comment]
@flavour Done the changes, Also added some more instances of start and end date.
Please review

**Comment by flavour** [comment]
Am still wary about looking for too many of these - rather keep it as a useful tool we can easily configure now.
If the issues highlighted are fixed then it can be merged, thanks :)

**Comment by gauravmittal1995** [comment]
@flavour  Done.

**Pull request merged**